azerothcore/azerothcore-wotlk
# Fix configs in docker builds

## Changes Proposed
- This fixes an issue where the .conf files aren't properly generated for the docker build
  - With this issue, the dbimport service in the docker compose file would fail to start due to insufficient config. 
  - related to PR #14707
- create a ".user" file extension for custom configs in `env/docker/etc/`
- Layers the ".dockerdist" file and the ".user" file over the ".dist" file (in that order), and then copies the ".dist" file to the base ".conf" file
- Docker specific config changes go in the ".dockerdist" file and should still be committed to the repo
- User specific config changes go in ".user" file and should not be committed to the repo
- If the ".dist" file and the base ".conf" file are different, the ".conf" will be backed up before the ".dist" file is copied into it

## Issues Addressed:

- There isn't a specific issue, however this was reported and discussed in discord over the course of yesterday and today

## SOURCE:


## Tests Performed:

- Executing commands to build and start the game server
- after creating an account, I can log in and play without any issues. 


## How to Test the Changes:


1.  Execute commands
   
    ```bash
    $ ./acore.sh docker clean:build && ./acore.sh docker build:nocache && ./acore.sh docker start:app
    ```
2. Create account and log in as normal

## Known Issues and TODO List:


- The largest issue that comes to mind is this is a breaking change for docker users (though, so was the change that caused this)
  - Docker users (and nobody else, of course) can no longer use the ".conf" for custom changes in docker setups. 
  - Their custom changes will have to be migrated to the ".user" file
  - Since the old ".conf" file is backed up, users should have minimal trouble copying the changes in the old ".conf" file to the new ".user" file


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

edit: link related PR
